### PR TITLE
Feature (eos_designs): max spines

### LINKED ISSUE #744: Changing amount of spines will renumber all leafs
## Issue Type

- [X] Bug Report
- [ ] EOS Configuration rendering error
- [ ] Documentation Issue Report

## Summary

When the number of Spines changes, all Leafs will renumber their loopback IPs (and router-ids, etc.). It seems that Leaf Loopback-IP is just `sum(Spines)+<Leaf id>`.

Maybe using an offset (Leaf Loopbacks / Router-IDs start at `<offset>` IPs after Spine IPs instead of directly after spine IPs) would help.

## Steps to reproduce

* Build config for fabric with 4 Spines
* Add two additional Spines
* See that all Leaf loopbacks got renumbered

### Expected results

Leafs would keep their loopback IP.

### Actual results

Leafs get renumbered, chaos and destruction follows.

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #744 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Add capability to defined `max_spines` variable to prevent re-addressing of the fabric if additional spine(s) are deployed after Day 2 operations.

## How to test

See molecule test results: evpn_underlay_ebgp_overlay_ebgp

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/fabric/bgp_overlay/switch-route-maps-evpn.j2 [hunk 1]
```diff
@@ -33,10 +33,10 @@
         type: permit
         set:
 {%     if leaf.mlag == true and leaf.mlag_role == "primary" %}
-          - "extcommunity soo {{ vtep_loopback_network_summary | ipaddr('network') | ipmath(leaf.id + spine.nodes | length) }}:1 additive"
+          - "extcommunity soo {{ vtep_loopback_network_summary | ipaddr('network') | ipmath(leaf.id + switch.max_spines) }}:1 additive"
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/fabric/bgp_overlay/switch-route-maps-evpn.j2:36]: Should not we protect using a var built in another logic file? If another designs leverage this file without computing `switch.max_spines` it won't work

Adding a header with variables required from `switch` namespace could help

**Review by mpergament — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md [hunk 2]
```diff
@@ -94,6 +94,8 @@ type: overlay-controller
 ```yaml
 # Defined in FABRIC.yml
 
+max_spines: < integer >= number of spine nodes | default spine.nodes | length >  # maximum number of spines, changing this parameter affects address allocation.
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md]: We could describe a use case where this knob could be used and not just default value.

**Review by titom73 — APPROVED** [review]
Looks good to me!
Just 2 comments for enhancements

**Review by titom73 — APPROVED** [review]
LGTM!

**Pull request merged**